### PR TITLE
past-session validator discovery APIs

### LINKED ISSUE #1461: Implement past-session validator discovery APIs for Network Bridge
#1452 introduced some APIs for the network bridge to enable validator discovery. Those APIs are open for discussion, but the information provided by them should stay pretty much the same: a way to signal to connect to particular validator IDs and learn the `PeerId`s associated with that set so as to be aware of when we are connected to those validators.

### PR DESCRIPTION
**Pull request by ordian**
Fixes #1461.

On top of #1968.

### CODE REVIEW

File: node/network/collator-protocol/src/collator_side.rs [hunk 1]
```diff
@@ -1085,25 +1072,40 @@ mod tests {
 			}
 		);
 
+		let current_index = 1;
```
Comment (coriolinus) [inline node/network/collator-protocol/src/collator_side.rs:1075]: A comment here would be nice, explaining why the current index is expected to be 1.

File: node/subsystem-util/src/lib.rs [hunk 2]
```diff
@@ -274,6 +275,7 @@ specialize_requests_ctx! {
 	fn request_validation_code_ctx(para_id: ParaId, assumption: OccupiedCoreAssumption) -> Option<ValidationCode>; ValidationCode;
 	fn request_candidate_pending_availability_ctx(para_id: ParaId) -> Option<CommittedCandidateReceipt>; CandidatePendingAvailability;
 	fn request_candidate_events_ctx() -> Vec<CandidateEvent>; CandidateEvents;
+	fn request_session_info_ctx(index: SessionIndex) -> Option<SessionInfo>; SessionInfo;
```
Comment (coriolinus) [inline node/subsystem-util/src/lib.rs:278]: Worth adding this to `specialize_requests!` also?

**Review by coriolinus — APPROVED** [review]
Only reviewed from 3d22556399a3d2fe64b4951e9703daeffcf7156a on; I believe that to be the start of this PR.

**Comment by ordian** [comment]
This PR removes `validator_discovery` runtime API call in favor of `session_info` (+ `session_index_for_child` to retrieve the current session index). I think it's redundant now, but if you disagree I can bring it back.

**Review by montekki — APPROVED** [review]

**Comment by montekki** [comment]
bot merge

**Pull request merged**